### PR TITLE
chore: Make SMTP environment variables configurable

### PR DESCRIPTION
**Pull request by sojan-official**
fixes: #1647

### CODE REVIEW

File: config/initializers/mailer.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  config.action_mailer.raise_delivery_errors = true
+
+  # Config related to smtp
+  smtp_settings = {
+    domain: ENV.fetch('SMTP_DOMAIN', ''),
+    address: ENV.fetch('SMTP_ADDRESS', 'localhost'),
+    port: ENV.fetch('SMTP_PORT', 587)
+  }
+
+  if ENV.fetch('SMTP_AUTHENTICATION', '').present?
```
Comment (pranavrajs) [inline config/initializers/mailer.rb:21]: Would this break the exisiting apps where they have not set SMTP_AUTHENTICATION? Also why is this under a condition?

Comment (sojan-official) [reply config/initializers/mailer.rb:21]: it only makes sense to pass authentication-related parameters to hash only when an authentication type is specified via `SMTP_AUTHENTICATION`.

in the current documentation even though we ask people to set this environment variable, the value for this is hardcoded in the production config. That's the reason why the SMTP server which has a different authentication mode than `login` isn't working

File: app/mailers/conversation_reply_mailer.rb [hunk 2]
```diff
@@ -105,15 +105,15 @@ def reply_email
     end
   end
 
-  def from_email_with_name
+  def from_email
     if should_use_conversation_email_address?
-      "#{assignee_name} <#{account_support_email}>"
```
Comment (pranavrajs) [inline app/mailers/conversation_reply_mailer.rb:110]: Why is the assignee_name removed?

File: app/mailers/conversation_reply_mailer.rb [hunk 3]
```diff
@@ -146,11 +146,7 @@ def current_domain
   end
 
   def account_support_email
```
Comment (pranavrajs) [inline app/mailers/conversation_reply_mailer.rb:148]: do we need this method?

File: config/initializers/mailer.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    smtp_settings[:user_name] = ENV.fetch('SMTP_USERNAME', '')
+    smtp_settings[:password] = ENV.fetch('SMTP_PASSWORD', '')
+    smtp_settings[:authentication] = ENV.fetch('SMTP_AUTHENTICATION', 'login')
```
Comment (pranavrajs) [inline config/initializers/mailer.rb:24]: I guess this should be a symbol.

**Pull request merged**